pkg/models/datasource_cache.go

```diff
@@ -201,22 +128,145 @@ func (ds *DataSource) GetHttpTransport() (*dataSourceTransport, error) {
 	return dsTransport, nil
 }
 
-func (ds *DataSource) sigV4Middleware(next http.RoundTripper) http.RoundTripper {
-	decrypted := ds.DecryptedValues()
+type proxyTransportCache2 struct {
+	cache map[int64]cachedRoundTripper2
+	sync.Mutex
+}
+
+type cachedRoundTripper2 struct {
+	updated      time.Time
+	roundTripper http.RoundTripper
+}
+
+var ptc2 = proxyTransportCache2{
+	cache: make(map[int64]cachedRoundTripper2),
+}
+
+func (ds *DataSource) GetHttpClient2(provider httpclient.Provider) (*http.Client, error) {
+	transport, err := ds.GetHttpTransport2(provider)
+	if err != nil {
+		return nil, err
+	}
+
+	return &http.Client{
+		Timeout:   ds.getTimeout(),
+		Transport: transport,
+	}, nil
+}
+
+func (ds *DataSource) GetHttpTransport2(provider httpclient.Provider) (http.RoundTripper, error) {
```

Comment:
This will be renamed before merge to `GetHttpTransport` and replace the existing `GetHttpTransport`